### PR TITLE
Add match and count APIs

### PR DESCRIPTION
**Pull request by ohltyler**
*Issue #, if available:*

*Description of changes:*

This PR adds frontend support for the implemented `match` and `count` APIs (see [here](https://github.com/opendistro-for-elasticsearch/anomaly-detection/pull/286)).

These apis are used in 2 different parts for validation on the create/edit detector page, replacing the search API which caused a security leak by returning extra detector information for a user which they may not be supposed to have access to.

Notes:
- confirmed no UT breaks
- confirmed both APIs return correct information
- tested on edit and create to confirm the validation for duplicate name worked appropriately

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by ylwu-amzn — APPROVED** [review]

File: server/routes/ad.ts [hunk 1]
```diff
@@ -735,6 +737,45 @@ const getAnomalyResults = async (
...
 };
 
+const matchDetector = async (
+  req: Request,
+  h: ResponseToolkit,
+  callWithRequest: CallClusterWithRequest
+): Promise<ServerResponse<AnomalyResults>> => {
+  try {
+    const { detectorName } = req.params;
+    console.log('calling matchDetector() w/ detector name: ', detectorName);
```
Comment (yizheliu-amazon) [inline server/routes/ad.ts:747]: extra log

File: server/routes/ad.ts [hunk 2]
```diff
@@ -735,6 +737,45 @@ const getAnomalyResults = async (
   }
 };
 
+const matchDetector = async (
```
Comment (yizheliu-amazon) [inline server/routes/ad.ts:740]: just curious: what does this API do? search detector by name?

File: server/routes/ad.ts [hunk 3]
```diff
@@ -735,6 +737,45 @@ const getAnomalyResults = async (
...
+    const response = await callWithRequest(req, 'ad.matchDetector', {
+      detectorName,
+    });
+    return {
+      ok: true,
+      response: response,
+    };
+  } catch (err) {
+    console.log('Anomaly detector - matchDetector', err);
+    return { ok: false, error: err.message };
```
Comment (yizheliu-amazon) [inline server/routes/ad.ts:757]: maybe use `getErrorMessage()` ?

File: server/routes/ad.ts (hunk 1 above)
Comment (ohltyler) [reply server/routes/ad.ts:747]: thanks, removed from `getDetectorCount()` too

**Comment by yizheliu-amazon** [comment]
can you quickly check if e2e test passes?

File: server/routes/ad.ts (hunk 3 above)
Comment (ohltyler) [reply server/routes/ad.ts:757]: good idea, will change

**Review by yizheliu-amazon — APPROVED** [review]

File: server/routes/ad.ts (hunk 2 above)
Comment (ohltyler) [reply server/routes/ad.ts:740]: basically returns `true` or `false` if the passed params match any existing detector name, without returning any extra detector information that a user may not have access to if using FGAC

Comment (ohltyler) [reply server/routes/ad.ts:740]: See backend PR for more https://github.com/opendistro-for-elasticsearch/anomaly-detection/pull/286

**Comment by ohltyler** [comment]
> can you quickly check if e2e test passes?

Getting a flaky test failing on my local, but these changes shouldn't affect it, I imagine it should still pass fine in the github runner

**Review by saratvemulapalli — APPROVED** [review]
Thanks Tyler for the change!

**Pull request merged**